doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -471,6 +471,39 @@ Field              | Type       | Meaning
 `cpu_nano_cores`   | [`bigint`] | Approximate CPU usage, in billionths of a vCPU core.
 `memory_bytes`     | [`bigint`] | Approximate RAM usage, in bytes.
 
+### `mz_storage_source_statistics`
+
+The `mz_storage_source_statistics` table gives the last known storage-specific statistics
+for all processes of all extant storage sources.
+
+At this time, we do not make any guarantees about the exactness or freshness of these numbers.
+
+Field                 | Type        | Meaning
+----------------------|-------------|--------
+`id`                 | [`text`]     | The ID of the storage object (source or sink).
+`worker_id`          | [`bigint`]   | An identifier of a process within a host.
+`snapshot_committed` | [`boolean`]  | Whether or not we have committed the initial snapshot for a source.
+`messages_received`  | [`bigint`]   | How many messages we have received from upstream.
+`messages_committed` | [`bigint`]   | How many messages we have committed into our storage layer.
+`bytes_received`     | [`bigint`]   | How many bytes worth of messages we have received from upstream.
+
+These values all require some amount of interpretation:
+
+- `messages_received` does not distinguish between inserts and deletes. Some upstream sources, like postgres,
+  produce deletion messages that we count the same as insertions.
+- `messages_committed` can be permanently lower or higher than `messages_received`, because:
+  - The storage layer can choose to consolidate messages within a certain time interval.
+  - Some envelopes (like `UPSERT`) can produce 2 messages (an insert and a delete) from a single upstream message.
+- `bytes_recieved` can represent different things for different sources.
+  - For example, it represents the literal size of the key and value from kafka messages, but represents
+  an internal representation of rows for postgres sources.
+- The `messages_*` and `bytes_received` metrics can be interpreted as rates that can loosely show
+  the changing speed of a source over time. These metrics are periodically reset to 0.
+- In general, the values should be summed across `worker_id`'s, and some metrics may only exist for some workers.
+  - Additionally, sources with multiple outputs (like postgres shards) will need `messages_committed`
+  and `snapshot_committed` summed/AND-ed across the top-level source and the connected tables.
```

Comment:
Thanks for taking the first cut at this. Looping in @morsapaes to make sure this goes through a docs style pass. The high-level point: this page, more than almost any other page in our docs, is reference documentation. It should deal almost exclusively in hard facts. We don't want to be giving folks advice about how to use the data here—that's better left to guides and tutorials—we should just describe what the data _is_.

Reply:
I definitely think this interpretation section is important, but if it should go somewhere else I'll move it!